src/sagemaker/huggingface/training_compiler/config.py

```diff
@@ -24,7 +26,14 @@
 class TrainingCompilerConfig(BaseConfig):
     """The SageMaker Training Compiler configuration class."""
 
-    SUPPORTED_INSTANCE_CLASS_PREFIXES = ["p3", "g4dn", "p4"]
+    SUPPORTED_INSTANCE_CLASS_PREFIXES = ["p3", "g4dn", "p4d", "g5"]
+    SUPPORTED_INSTANCE_TYPES_WITH_EFA = [
+        "ml.g4dn.8xlarge",
```

Comment:
current EFA is only exposed with `ml.p4d.24xlarge` & `ml.p3dn.24xlarge` for GPU instances, can we remove other GPU instances please. Please refer to https://aws.amazon.com/about-aws/whats-new/2021/05/amazon-sagemaker-supports-elastic-fabric-adapter-distributed-training/ for details.

Reply:
SMs contract is that if customer can use g5.48xl on EC2 with EFA for non-managed training experience, then a customer should be able to use same hardware features on SM with the managed experience.